src/pages/workspace/members/ImportedMembersPage.tsx

```diff
@@ -0,0 +1,167 @@
+import React, {useCallback, useState} from 'react';
+import {useOnyx} from 'react-native-onyx';
+import ConfirmModal from '@components/ConfirmModal';
+import HeaderWithBackButton from '@components/HeaderWithBackButton';
+import type {ColumnRole} from '@components/ImportColumn';
+import ImportSpreadsheetColumns from '@components/ImportSpreadsheetColumns';
+import ScreenWrapper from '@components/ScreenWrapper';
+import useLocalize from '@hooks/useLocalize';
+import {closeImportPage} from '@libs/actions/ImportSpreadsheet';
+import {importPolicyMembers} from '@libs/actions/Policy/Member';
+import Navigation from '@libs/Navigation/Navigation';
+import type {SettingsNavigatorParamList} from '@libs/Navigation/types';
+import CONST from '@src/CONST';
+import ONYXKEYS from '@src/ONYXKEYS';
+import ROUTES from '@src/ROUTES';
+import type SCREENS from '@src/SCREENS';
+
+type ImportedMembersPageProps = StackScreenProps<SettingsNavigatorParamList, typeof SCREENS.WORKSPACE.MEMBERS_IMPORTED>;
+
+function numberToColumn(index: number) {
+    let column = '';
+    let number = index;
+    while (number >= 0) {
+        column = String.fromCharCode((number % 26) + 65) + column;
+        number = Math.floor(number / 26) - 1;
+    }
+    return column;
+}
+
+function generateColumnNames(length: number) {
+    return Array.from({length}, (_, i) => numberToColumn(i));
+}
+
+function findDuplicate(array: string[]): string | null {
+    const frequencyCounter: Record<string, number> = {};
+
+    for (const item of array) {
+        if (item !== CONST.CSV_IMPORT_COLUMNS.IGNORE) {
+            if (frequencyCounter[item]) {
+                return item;
+            }
+            frequencyCounter[item] = (frequencyCounter[item] || 0) + 1;
+        }
+    }
+
+    return null;
+}
+
+function ImportedMembersPage({route}: ImportedMembersPageProps) {
+    const {translate} = useLocalize();
+    const [spreadsheet] = useOnyx(ONYXKEYS.IMPORTED_SPREADSHEET);
+    const [isImporting, setIsImporting] = useState(false);
+    const [isValidationEnabled, setIsValidationEnabled] = useState(false);
+    const policyID = route.params.policyID;
+    const columnNames = generateColumnNames(spreadsheet?.data?.length ?? 0);
+    const {containsHeader = true} = spreadsheet ?? {};
+
+    const columnRoles: ColumnRole[] = [
+        {text: translate('common.ignore'), value: CONST.CSV_IMPORT_COLUMNS.IGNORE},
+        {text: translate('common.email'), value: CONST.CSV_IMPORT_COLUMNS.EMAIL, isRequired: true},
+        {text: translate('common.role'), value: CONST.CSV_IMPORT_COLUMNS.ROLE},
+    ];
+
+    const requiredColumns = columnRoles.filter((role) => role.isRequired).map((role) => role);
+
+    const validate = useCallback(() => {
+        const columns = Object.values(spreadsheet?.columns ?? {});
+        let errors: Record<string, string | null> = {};
+
+        if (!requiredColumns.every((requiredColumn) => columns.includes(requiredColumn.value))) {
```

Comment:
Can we add a comment here? This code is really dense and hard to follow

Reply:
Yes, no problem; what do you think about this comment?